sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3942,6 +3943,69 @@ object functions {
    */
   def to_csv(e: Column): Column = to_csv(e, Map.empty[String, String].asJava)
 
+  // turn off style check that object names must start with a capital letter
+  // scalastyle:off
+  object partitioning {
```

Comment:
This seems breaking Maven build:

https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7/6803/console
https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7-ubuntu-testing/1712/consoleFull
https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2/289/consoleFull
https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2-jdk-11/357/consoleFull

```
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:32: cannot access org.apache.spark.sql.functions.1
  class file for org.apache.spark.sql.functions$1 not found
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:70: cannot find symbol
  symbol:   method avg(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:81: cannot find symbol
  symbol:   method col(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:89: cannot find symbol
  symbol:   method col(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:90: cannot find symbol
  symbol:   method col(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:91: cannot find symbol
  symbol:   method col(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
[ERROR] [Error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-testing/sql/hive/src/test/java/org/apache/spark/sql/hive/JavaDataFrameSuite.java:92: cannot find symbol
  symbol:   method col(java.lang.String)
  location: class org.apache.spark.sql.hive.JavaDataFrameSuite
```

Seems after this commit, Github workflow also seems causing build failures:

![Screen Shot 2019-09-02 at 12 40 04 PM](https://user-images.githubusercontent.com/6477701/64088987-e1d64480-cd7e-11e9-9b45-d34ed9eab7f4.png)